### PR TITLE
docs: mark the v1 line as EOL

### PR DESCRIPTION
**Pull request by mpoke**
## Description

Closes: NA

- [x] Update compatibility matrix @bermuell 
- also mark `v2.1.0-provider-lsm` as EOL

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct `docs:` prefix in the PR title
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] reviewed "Files changed" and left comments if necessary 
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct `docs:` prefix in the PR title
- [ ] Confirmed all author checklist items have been addressed 
- [ ] Confirmed that this PR only changes documentation
- [ ] Reviewed content for consistency
- [ ] Reviewed content for spelling and grammar
- [ ] Tested instructions (if applicable)
- [ ] Checked that the documentation website can be built and deployed successfully (run `make build-docs`)

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -94,24 +92,22 @@ Versions of Golang, IBC, Cosmos SDK and CometBFT used by ICS in the currently ac
...
+| `v4.0.0` || ✅ (3) | ✅ (3) | ✅ (3) | ✅ (3) | ✅ (3)| ✅ (3)| ✅ |
 
 #### Notes
 
-The following adjustments must be made to the CCV consumer genesis state that is obtained from the provider chain after the spawn time is reached in order for the consumer chain to start without errors. 
+The following adjustments must be made to the CCV consumer genesis state that is obtained from the provider chain after the spawn time is reached in order for the consumer chain to start without errors.
 
 - (1) Remove “preCCV” key
 - (2) Remove “prehash_key_before_comparison” keys
-- (3) Add .params.retry_delay_period: “3600s”
```
Comment (bermuell) [inline RELEASES.md:116]: Removed the `retry_delay_period` as it should be taken care by the `genesis transform` cli.
To be discussed if this hides some internals from someone who planned to do the adjustments by hand.

Comment (mpoke) [reply RELEASES.md:116]: cc @dasanchez

File: RELEASES.md [hunk 2]
```diff
@@ -9,109 +9,102 @@
...
+| `v2.0.0` || ✅ | ✅ | ✅ (1) | ✅ (1) | ✅ (1),(2) | ❌ |
+| `v3.1.0` || ✅ | ✅ | ✅ | ✅ | ✅ (2) | ❌ |
+| `v3.2.0` || ✅ | ✅ | ✅ | ✅ | ✅ (2) | ✅ |
+| `v3.3.0` || ✅ (2) | ✅ (2) | ✅ (2) | ✅ (2) | ✅ | ✅ |
+| `v4.0.0` || ✅ (2) | ✅ (2) | ✅ (2) | ✅ (2)| ✅ (2)| ✅ |
 
 #### Notes
 
-The following adjustments must be made to the CCV consumer genesis state that is obtained from the provider chain after the spawn time is reached in order for the consumer chain to start without errors. 
+The following adjustments must be made to the CCV consumer genesis state that is obtained from the provider chain after the spawn time is reached in order for the consumer chain to start without errors.
```
Comment (insumity) [inline RELEASES.md:107]: Not specific to this PR but not clear to me the context of those `Notes` here. Are the notes intended for when someone is planning to use a previous MAJOR version of ICS or is it more general?

**Review by insumity — APPROVED** [review]
LGTM

I think it would be nice to mention in the description of this PR on why `v1.x` and `v2.1.0-provider-lsm` are marked End Of Life (EOL) when their respective [EOL dates](https://github.com/cosmos/interchain-security/blob/main/RELEASES.md#stable-release-policy) are February 21, 2024 and June 09, 2024 respectively. I would assume that [would be](https://github.com/cosmos/interchain-security/blob/main/RELEASES.md#stable-release-policy):
>If there is no known chain using a major release series, then it reached end of life. This is a temporary policy that is possible due to the relatively low number of consumer chains.

?

**Comment by mpoke** [comment]
> I think it would be nice to mention in the description of this PR on why `v1.x` and `v2.1.0-provider-lsm` are marked End Of Life (EOL) when their respective [EOL dates](https://github.com/cosmos/interchain-security/blob/main/RELEASES.md#stable-release-policy) are February 21, 2024 and June 09, 2024 respectively. I would assume that [would be](https://github.com/cosmos/interchain-security/blob/main/RELEASES.md#stable-release-policy):
> 
> > If there is no known chain using a major release series, then it reached end of life. This is a temporary policy that is possible due to the relatively low number of consumer chains.

Exactly. We need to reduce the amount of active releases that we maintain, especially as we need to test compatibility between them.

File: RELEASES.md (hunk 2 above)
Comment (mpoke) [reply RELEASES.md:107]: The notes are intended for using two different versions of ICS on the consumer and provider.

**Review by bermuell — APPROVED** [review]

**Pull request merged**